test/Microsoft.ML.StaticPipelineTesting/TreeRepresentation.cs

```diff
@@ -0,0 +1,89 @@
+﻿using Microsoft.ML.Data;
+using Microsoft.ML.RunTests;
+using Microsoft.ML.StaticPipe;
+using Microsoft.ML.Trainers.FastTree;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.ML.StaticPipelineTesting
+{
+    public sealed class TreeRepresentation : BaseTestClassWithConsole
+    {
+        public TreeRepresentation(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact]
+        public void FastTreeRegressionRepresentation()
+        {
+            var env = new MLContext(seed: 0);
+            var dataPath = GetDataPath(TestDatasets.generatedRegressionDataset.trainFilename);
+            var dataSource = new MultiFileSource(dataPath);
+
+            var catalog = new RegressionCatalog(env);
+
+            var reader = TextLoaderStatic.CreateReader(env,
+                c => (label: c.LoadFloat(11), features: c.LoadFloat(0, 10)),
+                separator: ';', hasHeader: true);
+
+            FastTreeRegressionModelParameters pred = null;
+
+            var est = reader.MakeNewEstimator()
+                .Append(r => (r.label, score: catalog.Trainers.FastTree(r.label, r.features,
+                    numTrees: 10,
+                    numLeaves: 5,
+                    onFit: (p) => { pred = p; })));
```

Comment:
My understanding is that we are no longer doing any development on Static API. Do we need test using Static API?

Reply:
This test is not for static API, it's for tree's public area.

---
In reply to: [252028640](https://github.com/dotnet/machinelearning/pull/2243#discussion_r252028640) [](ancestors = 252028640)

Reply:
Yes, but it used Static API and I think you should write test using dynamic API (using MLContext) instead of Static API. This is what we decided for going forward.

---
In reply to: [252031278](https://github.com/dotnet/machinelearning/pull/2243#discussion_r252031278) [](ancestors = 252031278,252028640)

Reply:
As long as they are testing the same thing, I don't see a reason to create one extra test.

---
In reply to: [252036849](https://github.com/dotnet/machinelearning/pull/2243#discussion_r252036849) [](ancestors = 252036849,252031278,252028640)

Reply:
I am not asking for write another test. I saying this test should use Dynamic API.

---
In reply to: [252040126](https://github.com/dotnet/machinelearning/pull/2243#discussion_r252040126) [](ancestors = 252040126,252036849,252031278,252028640)

Reply:
I don't think this is the right way to do the test here. We don't expose ModelParameters through dynamic in this specific case to users.

---
In reply to: [252041372](https://github.com/dotnet/machinelearning/pull/2243#discussion_r252041372) [](ancestors = 252041372,252040126,252036849,252031278,252028640)

Reply:
can you see this test
[https://github.com/dotnet/machinelearning/blob/e3830910531f00013c27391914233a085a1394a4/test/Microsoft.ML.Tests/TrainerEstimators/TreeEstimators.cs#L28](https://github.com/dotnet/machinelearning/blob/e3830910531f00013c27391914233a085a1394a4/test/Microsoft.ML.Tests/TrainerEstimators/TreeEstimators.cs#L28)

The only thing changes in your case is line 31-35

---
In reply to: [252055619](https://github.com/dotnet/machinelearning/pull/2243#discussion_r252055619) [](ancestors = 252055619,252041372,252040126,252036849,252031278,252028640)

Reply:
We should then file a bug that its not possible do this scenario with dynamic API.

---
In reply to: [252060908](https://github.com/dotnet/machinelearning/pull/2243#discussion_r252060908) [](ancestors = 252060908,252059592,252055619,252041372,252040126,252036849,252031278,252028640)